static/app/views/eventsV2/resultsChart.tsx

```diff
@@ -158,6 +159,19 @@ type ContainerProps = {
 };
 
 class ResultsChartContainer extends Component<ContainerProps> {
+  state = {
+    yAxisOptions: this.getYAxisOptions(this.props.eventView),
+  };
+
+  componentWillReceiveProps(nextProps) {
+    const yAxisOptions = this.getYAxisOptions(this.props.eventView);
+    const nextYAxisOptions = this.getYAxisOptions(nextProps.eventView);
+
+    if (!valueIsEqual(yAxisOptions, nextYAxisOptions, true)) {
+      this.setState({yAxisOptions: nextYAxisOptions});
+    }
+  }
```

Comment:
The y-axis menu re-renders every time `ResultsChartContainer` re-renders (which happens a lot, e.g. every time the selection value changes), causing the focus state to reset:
https://user-images.githubusercontent.com/44172267/167709646-12671188-9438-41f5-a177-f32a2e9b6def.mov

This happens because the `yAxisOptions` array is defined inside the `render` function. To fix this problem, we can store `yAxisOptions` as a state that only gets updated when it actually changes, ignoring all other prop updates.